php/class-ee-site.php

```diff
@@ -0,0 +1,399 @@
+	}
+
+	/**
+	 * Function to delete the given site.
+	 *
+	 * @param int $level
+	 *  Level of deletion.
+	 *  Level - 0: No need of clean-up.
+	 *  Level - 1: Clean-up only the site-root.
+	 *  Level - 2: Try to remove network. The network may or may not have been created.
+	 *  Level - 3: Disconnect & remove network and try to remove containers. The containers may not have been created.
+	 *  Level - 4: Remove containers.
+	 *  Level - 5: Remove db entry.
+	 *
+	 * @ignorecommand
+	 */
+	public function delete_site( $level, $site_name, $site_root ) {
+		$this->fs   = new Filesystem();
+		$proxy_type = EE_PROXY_TYPE;
+		if ( $level >= 3 ) {
+			if ( EE::docker()::docker_compose_down( $site_root ) ) {
+				EE::log( "[$site_name] Docker Containers removed." );
+			} else {
+				\EE\Utils\default_launch( "docker rm -f $(docker ps -q -f=label=created_by=EasyEngine -f=label=site_name=$site_name)" );
+				if ( $level > 3 ) {
+					EE::warning( 'Error in removing docker containers.' );
+				}
+			}
+
+			EE::docker()::disconnect_site_network_from( $site_name, $proxy_type );
+		}
+
+		if ( $level >= 2 ) {
+			if ( EE::docker()::rm_network( $site_name ) ) {
+				EE::log( "[$site_name] Docker container removed from network $proxy_type." );
+			} else {
+				if ( $level > 2 ) {
+					EE::warning( "Error in removing Docker container from network $proxy_type" );
+				}
+			}
+		}
+
+		if ( $this->fs->exists( $site_root ) ) {
+			try {
+				$this->fs->remove( $site_root );
+			}
+			catch ( Exception $e ) {
+				EE::debug( $e );
+				EE::error( 'Could not remove site root. Please check if you have sufficient rights.' );
+			}
+			EE::log( "[$site_name] site root removed." );
+		}
+
+		if ( $level > 4 ) {
+			if ( $this->le ) {
+				EE::log( 'Removing ssl certs.' );
+				$crt_file   = EE_CONF_ROOT . "/nginx/certs/$site_name.crt";
+				$key_file   = EE_CONF_ROOT . "/nginx/certs/$site_name.key";
+				$conf_certs = EE_CONF_ROOT . "/acme-conf/certs/$site_name";
+				$conf_var   = EE_CONF_ROOT . "/acme-conf/var/$site_name";
+
+				$cert_files = [ $conf_certs, $conf_var, $crt_file, $key_file ];
+				try {
+					$this->fs->remove( $cert_files );
+				}
+				catch ( Exception $e ) {
+					EE::warning( $e );
+				}
+
+			}
+			if ( EE::db()::delete( array( 'sitename' => $site_name ) ) ) {
+				EE::log( 'Removing database entry.' );
+			} else {
+				EE::error( 'Could not remove the database entry' );
+			}
+		}
+		EE::log( "Site $site_name deleted." );
+	}
+
+	/**
+	 * Enables a website. It will start the docker containers of the website if they are stopped.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [<site-name>]
+	 * : Name of website to be enabled.
+	 */
+	public function up( $args, $assoc_args ) {
+		\EE\Utils\delem_log( 'site enable start' );
+		$args = \EE\SiteUtils\auto_site_name( $args, 'site', __FUNCTION__ );
+		$this->populate_site_info( $args );
+		EE::log( "Enabling site $this->site_name." );
+		if ( EE::docker()::docker_compose_up( $this->site_root ) ) {
```

Comment:
Also we need to check which services are enabled. We can't just do a `docker-compose up -d`

Reply:
@mbtamuli definitely, can't simply run `docker-compose up -d`. This has been handled and taken care in a separate PR where I am dealing with services.

There is functionality to run only `docker-compose up -d [required containers]` https://github.com/EasyEngine/site-command/pull/72/files#diff-8d90b355b432fc2c35f58491fcf26ae3R355

Only the services enabled by the user for that specific site will be enabled on running `up()`.

Reply:
@rahulsprajapati update the `up()` function. Added a check to see if the site is already enabled or not. Printing error message if a site is already enabled. 

Also, just in case someone does want to force re-up the site for some reason, I have added a `--force` parameter which will enable an option to run `docker-compose up -d` for the sites containers even when the site is enabled.